include/decision_response.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <cstddef>
+#include <iterator>
+#include <vector>
+
+#include "ranking_response.h"
+
+namespace reinforcement_learning {
+  class api_status;
+
+  class decision_response {
+  public:
+    using coll_t = std::vector<ranking_response>;

```

Comment:
In ranking_response, we moved away from stl collections in the public interface due to concerns that collections that might be compiled with different flags.  This resulted in defining iterators independely of stl.